src/libsystemd/sd-bus/sd-bus.c

```diff
@@ -4121,35 +4118,34 @@ _public_ int sd_bus_path_decode_many(const char *path, const char *path_template
 
         for (template_pos = path_template; *template_pos; ) {
                 const char *sep;
-                size_t length;
+                size_t length, path_length;
                 char *label;
 
                 /* verify everything until the next '%' matches verbatim */
                 sep = strchrnul(template_pos, '%');
                 length = sep - template_pos;
-                if (strncmp(path_pos, template_pos, length))
+                if (!strneq(path_pos, template_pos, length))
                         return 0;
 
                 path_pos += length;
-                template_pos += length;
 
-                if (!*template_pos)
+                if (!*sep)
                         break;
 
                 /* We found the next '%' character. Everything up until here
                  * matched. We now skip ahead to the end of this label and make
                  * sure it matches the tail of the label in the path. Then we
                  * decode the string in-between and save it for later use. */
 
-                ++template_pos; /* skip over '%' */
+                template_pos = sep + 1; /* skip over '%' */
```

Comment:
Please drop the change. `path_pos` and `template_pos` are both kind of cursors on `path` and `template`, and these should be handled in the same way.